woocommerce/woocommerce-ios
# Product Creation with AI v2 - Milestone 3: progress bar and label with dynamic color and text + logic

Part of #13274
Closes #13111 and decision-made based on C03L1NF1EA3/p1720512584126239

I'm also waiting for a feedback here C03L1NF1EA3/p1720515544395649

## Description
As part of Milestone 3, I implemented this new view with a progress bar and a label that changes status based on the number of words the user is typing in the starting information screen of Product Creation with AI.
This feature is hidden under a feature flag.

## Changes:
- `ProductCreationAIBarProgressStyle`: created a new ProgressViewStyle that match the style proposed in the Figma design, for the progress bar.
- `ProductCreationAIPromptProgressBar`: the view containing the progress bar, the text and the logic.
- `ProductCreationAIPromptProgressBarViewModel` + unit tests: view model of `ProductCreationAIPromptProgressBar`, containing the logic of the progress bar and the various sttuses. Right now we have 5 statuses matching the Figma design, and the range of words for matching the various statuses.

## Testing information
1. Start creating a new product with AI, and start typing.
2. Check the progress bar color and progress and how the suggestion text changes based on the number of words added.
3. Disable the feature flag `productCreationAIv2M3`, then check that the progress view is hidden.

## Demo
https://github.com/woocommerce/woocommerce-ios/assets/495617/6a4149ea-e54e-4535-baa4-5f9b851520b1



---
